src/tess_asteroids/tpf.py

```diff
@@ -356,17 +357,115 @@ def _bg_rolling_median(self, nframes: int = 25):
 
         return np.asarray(bg), np.asarray(bg_err)
 
-    def get_aperture(self):
+    def create_aperture(self, method: str = "threshold", **kwargs):
+        """
+        Creates an aperture mask using a method ['threshold', 'prf', 'ellipse'].
+        It creates the `self.aperture_mask` attribute with the 3D mask.
+
+        Parameters
+        ----------
+        method : str
+            Method used for apertrue estimation. One of ['threshold', 'prf', 'ellipse'].
+        kwargs : dict
+            Keywords arguments passed to aperture mask method, e.g
+            `self._create_threshold_mask` takes `threshold` and `reference_pixel`.
+
+        Returns
+        -------
+        """
+        if not hasattr(self, "all_flux") or not hasattr(self, "flux"):
+            raise AttributeError(
+                "Must run `get_data()` and `reshape_data()` before computing aperture mask."
+            )
+
+        # Get mask via chosen method
+        if method == "threshold":
+            self.aperture_mask = self._create_threshold_mask(**kwargs)
+        # will add these methods in a future PR
+        elif method in ["prf", "ellipse"]:
+            raise NotImplementedError(f"Method '{method}' not implemented yet.")
+        else:
+            raise ValueError(
+                f"`{method}` must be one of: ['threshold', 'prf', 'ellipse']"
+            )
+
+    def _create_threshold_mask(
+        self,
+        threshold: float = 3.0,
+        reference_pixel: Union[str, Tuple[float, float]] = "center",
+    ):
         """
         Get aperture for LC extraction.
 
         Parameters
         ----------
+        threshold : float
+            A value for the number of sigma by which a pixel needs to be
+            brighter than the median flux to be included in the aperture mask.
+        reference_pixel: (int, int) tuple, 'center', or None
+            (row, column) pixel coordinate closest to the desired region.
+            For example, use `reference_pixel=(0,0)` to select the region
+            closest to the bottom left corner of the target pixel file.
+            If 'center' (default) then the region closest to the center pixel
+            will be selected. If `None` then all regions will be selected.
 
         Returns
         -------
+        aperture_mask : ndarray
+            Boolean numpy array containing `True` for pixels above the
+            threshold. Shape is (ntimes, nrows, ncols)
         """
-        raise NotImplementedError("get_aperture() is not yet implemented.")
+        if (
+            not hasattr(self, "all_flux")
+            or not hasattr(self, "flux")
+            or not hasattr(self, "corr_flux")
+        ):
+            raise AttributeError(
+                "Must run `get_data()`, `reshape_data()` and `background_correction()` before."
+            )
+
+        if reference_pixel == "center":
+            reference_pixel = (self.shape[1] / 2, self.shape[0] / 2)
```

Comment:
```suggestion
            reference_pixel = ((self.shape[0] - 1) / 2, (self.shape[1] - 1) / 2)
```
Needs to be row,col and must subtract one to get correct midpoint.